File: 55_jump_game/canJump.go
func canJump(nums []int) bool {
	r, i, n := 0, 0, len(nums)

	for i = 0; i < n && i <= r; i++ {
		r = findMax(r, i+nums[i])
	}

	return i == n
}

func findMax(a int, b int) int {
	if a > b {
		return a
	} else {
		return b
	}
}

File: 55_jump_game/canJump.java
class Solution {
    public boolean canJump(int[] nums) {
        int reach = 0;
        int n = nums.length;
        int i;
        
        for (i = 0; i < n && i <= reach; i++) {
            reach = Math.max(reach, i + nums[i]);
        }

        return i == n;
    }
}